help_function/src/onedpl_test_group_load.cpp

```diff
@@ -0,0 +1,126 @@
+            }
+        });
+  });
+  q.wait_and_throw();
+  
+  sycl::host_accessor data_accessor(buffer, sycl::read_only);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_validation_function(ptr);
+}
+
+bool test_load_subgroup_striped_standalone() {
+  sycl::queue q;
+  int data[512];
+  for (int i = 0; i < 512; i++) data[i] = i;
+
+  sycl::buffer<int, 1> buffer(data, 512);
+  q.submit([&](sycl::handler &h) {
+    sycl::accessor dacc(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d = dacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          dpct::group::uninitialized_load_subgroup_striped<128>(item, d, thread_data);
+          // Write thread_data of each work item at index to the global buffer
+          int global_index = item.get_global_linear_id() * 4; // Each thread_data has 4 elements
+            for (int i = 0; i < 4; ++i) {
+                buffer[global_index + i] = thread_data[i];
+            }
+        });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer, sycl::read_only);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_validation_function(ptr);
+}
+
+bool test_load_blocked_striped_standalone(dpct::group::load_algorithm T) {
+  sycl::queue q;
+  int data[512];
+  for (int i = 0; i < 512; i++) data[i] = i;
+
+  sycl::buffer<int, 1> buffer(data, 512);
+  q.submit([&](sycl::handler &h) {
+    sycl::accessor dacc(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d = dacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          dpct::group::load_blocked<128, T>(item, d, thread_data);
+          // Write thread_data of each work item at index to the global buffer
+          int global_index = item.get_global_linear_id() * 4; // Each thread_data has 4 elements
+            for (int i = 0; i < 4; ++i) {
+                buffer[global_index + i] = thread_data[i];
+            }
+        });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer, sycl::read_only);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_validation_function(ptr);
+}
+
+int main() {
+  return !(test_load_blocked_striped(dpct::group::load_algorithm::BLOCK_LOAD_DIRECT) && test_load_blocked_striped(dpct::group::load_algorithm::BLOCK_LOAD_STRIPED) && test_load_subgroup_striped_standalone()
+           && test_load_blocked_striped_standalone(dpct::group::load_algorithm::BLOCK_LOAD_STRIPED) && test_load_blocked_striped_standalone(dpct::group::load_algorithm::BLOCK_LOAD_DIRECT));

```

Comment:
Can you expand the main to provide descriptive messages when specific tests fail which include the load strategy being tested, and collect their result in some variable which counts test failures, and return that.  
Look at some of the other tests in this repository and try to use a similar structure.